test/infrastructure/docker/docker/util.go

```diff
@@ -93,43 +91,48 @@ func getContainer(filters ...string) (*types.Node, error) {
 // List returns the list of container IDs for the kind "nodes", optionally
 // filtered by docker ps filters
 // https://docs.docker.com/engine/reference/commandline/ps/#filtering
-func List(filters ...string) ([]*types.Node, error) {
+func List(filters Filter) ([]*types.Node, error) {
 	res := []*types.Node{}
 	visit := func(cluster string, node *types.Node) {
 		res = append(res, node)
 	}
-	return res, list(visit, filters...)
+	return res, list(visit, filters)
 }
 
-func list(visit func(string, *types.Node), filters ...string) error {
-	args := []string{
-		"ps",
-		"-q",         // quiet output for parsing
-		"-a",         // show stopped nodes
-		"--no-trunc", // don't truncate
-		// filter for nodes with the cluster label
-		"--filter", "label=" + clusterLabelKey,
-		// format to include friendly name and the cluster name
-		"--format", fmt.Sprintf(`{{.Names}}\t{{.Label "%s"}}\t{{.Image}}\t{{.Status}}`, clusterLabelKey),
+func list(visit func(string, *types.Node), filters Filter) error {
+	ctx := context.Background()
```

Comment:
maybe we can also pass through the context to here? If not, afaik `context.TODO()` is more appropriate here.

Reply:
Yeah, this one looks a little trickier since it's buried under several different code paths that don't currently pass a context. I'll make `TODO` for now and see if we can clean that up when moving to an interface.

Reply:
Absolutely fine for me